webui/react/src/hooks/useModal/useModal.tsx

```diff
@@ -106,10 +107,19 @@ const useModal = (
     }
   }, [ extendEventHandler, modalProps, options, prevModalProps ]);
 
+  // Sets componentUnmounting to true only when the parent component is unmounting so that the next
+  // useEffect only runs modalClose on unmount, rather than every time modalClose updates.
+  // The order of these two useEffects matters, this one has to be first.
```

Comment:
great explanation for this!

Reply:
**nit:** for multi-line comments, suggest the use of `/** */` format:
```
/**
 * Sets componentUnmounting to true only when the parent component is unmounting so that the next
 * useEffect only runs modalClose on unmount, rather than every time modalClose updates.
 * The order of these two useEffects matters, this one has to be first.
 */
```